engine/title.asm

```diff
@@ -3,17 +3,654 @@ INCLUDE "constants.asm"
+	ld de, $0001
+	call PlayMusic		; Play Title Theme
+	ret
+
+InitFlashTitle:: ;6087
+	call IncreaseSequenceNumber
+	ld a, $1A
+	ld [wSequence+2], a
+	ld a, $06
+	ld [wSequence+3], a
+	ret
+
+FlashTitle:: ;6095
+	ld hl, wSequence+3
+	ld a, [hl]
+	and a
+	jr z, .exit
+	dec [hl]
+	ld a, [wSequence+2]
+	xor $1A
+	ld [wSequence+2], a
+	ldh [rBGP], a
+	call DelayFrame
+	call DelayFrame
+	ret
+
+.exit
+	call IncreaseSequenceNumber
+	ld a, $E4
+	ldh [rBGP], a
+	ret
+
+PMJapaneseChara:: ;60b6
+	call PrintPMJapaneseChara
+	ld a, $10
+	ld [wSequence+2], a
+	call IncreaseSequenceNumber
+	ld a, $01
+	ldh [$FFDE], a
+	ret
+
+PMSubtitle:: ;60c6
+	call PrintPMSubtitle
+	ld a, $10
+	ld [wSequence+2], a
+	call IncreaseSequenceNumber
+	ld a, $01
+	ldh [$FFDE], a
+	ret
+
+Version:: ;60d6
+	call PrintVersion
+	ld a, $10
+	ld [wSequence+2], a
+	call IncreaseSequenceNumber
+	ld a, $01
+	ldh [$FFDE], a
+	ret
+
+CopyRight:: ;60e6
+	call PrintCopyRight
+	ld a, $10
+	ld [wSequence+2], a
+	call IncreaseSequenceNumber
+	ld a, $01
+	ldh [$FFDE], a
+	ret
+
+HoOh:: ;60f6
+	call SetHoOh
+	ld a, $10
+	ld [wSequence+2], a
+	call IncreaseSequenceNumber
+	ld a, $01
+	ldh [$FFDE], a
+	ret
+
+PressButtonInit:: ;6106
+	ld hl, wSequence
+	inc [hl]
+	ld hl, wSequence+2
+	ld de, DecodeNybble0Table - 3	; DecodeNybble0Table - 3 = $0C00
+	ld [hl], e
+	inc hl
+	ld [hl], d
+	ret	
+	
+TitleScreenInputAndTimeout:: ;6114
+	ld hl, wSequence+2
+	ld e, [hl]
+	inc hl
+	ld d, [hl]
+	ld a, e
+	or d
+	jr z, .psbtn_reset
+	dec de
+	ld [hl], d
+	dec hl
+	ld [hl], e
+	call GetJoypad
+	ld hl, hJoyState	; hJoyState = $FFA3
+	ld a, [hl]
+	and $46				; UP + B + SELECT brings you to the SRAM clear screen.
+	cp $46
+	jr z, .psbtn_sramclear
+	ld a, [hl]
+	and $04				; SELECT will bring you to the debug menu.
+	jr nz, .psbtn_gotodebug
+	ld a, [hl]
+	and $09
+	ret z
+	
+.psbtn_play
+	ld a, $00			;MainMenu
+	jr .psbtn_nextseq
+
+.psbtn_gotodebug
+if DEBUG 
+	ld a, $01			;DebugMenu
+	jr .psbtn_nextseq
+else
+	ret
+endc
+
+.psbtn_sramclear
+	ld a, $02			;SRAMClearMenu
+	
+.psbtn_nextseq
+	ld [wSequence + 1], a	; wSequence + 1 = $CB5F
+	ld hl, wSequence	; wSequence = $CB5E
+	set 7, [hl]
+	ret
+
+.psbtn_reset
+	ld hl, wSequence	; wSequence = $CB5E
+	inc [hl]
+	xor a
+	ld [wNextBGM], a ;c1a7
+	ld [wNextBGM + 1], a ;c1a8
+	ld hl, wSoundFade ;c1a5
+	ld [hl], $08
+	ret	
+	
+FadeMusicOut:: ;615c
+	ld a, [wSoundFade] ;c1a5
+	and a
+	ret nz
+	ld a, $03
+	ld [wSequence + 1], a
+	ld hl, wSequence
+	set 7, [hl]
+	ret	
+
+SetLYOverrides:: ;616c
+	ld hl, wLYOverrides
+	ld c, $30
+.setly_loop
+	ldi [hl], a
+	dec c
+	jr nz, .setly_loop
+	ret
+
+PrintPMSubtitle:: ;6176
+	;ld hl, _RAM_C31A_ ;c31a
+	coord hl, 2, 6
+	ld b, $0F
+	ld a, $69
+	jr LoadPrintArea ;6186
+
+PrintVersion:: ;617f
+	;ld hl, _RAM_C2B8_ ;c2b8
+	coord hl, 4, 1
+	ld b, $09
+	ld a, $60
+	
+LoadPrintArea:: ;6186
+	ldi [hl], a
+	inc a
+	dec b
+	jr nz, LoadPrintArea
+	ret
+
+PrintPMJapaneseChara:: ;618C
+	;ld hl, _RAM_C2D7_
+	coord hl, 15, 2
+	ld a, $BA
+	ld bc, $0404
```

Comment:
`lb bc, 4, 4`. We know this uses a `ld bc` to load two 8-bit values into `b` and `c` because of the way the registers are used in the following function (i.e. `PrintBoxArea` uses `b` and `c` as two counters).

Reply:
Done.